From wfp5p@virginia.edu  Thu Jul  2 12:52:23 2009
From: Bill Pemberton <wfp5p@virginia.edu>
Date: Fri, 19 Jun 2009 12:32:56 -0400
Subject: Staging: serqt_usb2: fix qt_close parameters in serqt_usb2
To: greg@kroah.com
Cc: Bill Pemberton <wfp5p@virginia.edu>
Message-ID: <1245429177-7956-2-git-send-email-wfp5p@virginia.edu>


The parameter list for qt_close() was from the old non usb-serial
driver.

Signed-off-by: Bill Pemberton <wfp5p@virginia.edu>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/serqt_usb2/serqt_usb2.c |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- a/drivers/staging/serqt_usb2/serqt_usb2.c
+++ b/drivers/staging/serqt_usb2/serqt_usb2.c
@@ -1041,17 +1041,19 @@ static void qt_block_until_empty(struct 
 	}
 }
 
-static void qt_close(struct tty_struct *tty, struct usb_serial_port *port,
-		     struct file *filp)
+static void qt_close( struct usb_serial_port *port)
 {
 	struct usb_serial *serial = port->serial;
 	struct quatech_port *qt_port;
 	struct quatech_port *port0;
+	struct tty_struct *tty;
 	int status;
 	unsigned int index;
 	status = 0;
 
 	dbg("%s - port %d\n", __func__, port->number);
+
+	tty = tty_port_tty_get(&port->port);
 	index = tty->index - serial->minor;
 
 	qt_port = qt_get_port_private(port);
